lib/export-scala-base.js

```diff
@@ -334,8 +342,45 @@ ${
   )
 }
 
-class N${comp.name}TopBase(c: N${comp.name}TopParams)(implicit p: Parameters) extends SimpleLazyModule {
+
+class N${comp.name}TopLogicalTreeNode(component: N${comp.name}TopBase) extends LogicalTreeNode(() => Some(component.imp.device)) {
+  override def getOMComponents(resourceBindings: ResourceBindings, components: Seq[OMComponent]): Seq[OMComponent] = {
+    DiplomaticObjectModelAddressing.getOMComponentHelper(
+      resourceBindings, (resources) => {
+        val omDevice = new scala.collection.mutable.LinkedHashMap[String, Any] with OMDevice {
+          val memoryRegions: Seq[OMMemoryRegion] = Nil // TODO
+          val interrupts: Seq[OMInterrupt] = Nil // TODO
```

Comment:
Is your plan to fill in this in this PR, or did you want to split it out into a separate one?

Reply:
just added these